src/Compilers/CSharp/Portable/Compiler/Instrumentation.cs

```diff
@@ -199,11 +199,52 @@ public override BoundNode Visit(BoundNode node)
                         }
                         break;
                     case BoundKind.LocalDeclaration:
-                        if (statement.Syntax.Parent.Kind() == SyntaxKind.VariableDeclaration && ((VariableDeclarationSyntax)statement.Syntax.Parent).Variables.Count > 1)
+                        if (statement.Syntax.Parent.Kind() == SyntaxKind.VariableDeclaration)
                         {
-                            // This declaration is part of a MultipleLocalDeclarations statement,
-                            // and so should not be treated as a separate statement.
-                            return visited;
+                            VariableDeclarationSyntax declarationSyntax = (VariableDeclarationSyntax)statement.Syntax.Parent;
+                            if (declarationSyntax.Variables.Count > 1)
+                            {
+                                // This declaration is part of a MultipleLocalDeclarations statement,
+                                // and so should not be treated as a separate statement.
+                                return visited;
+                            }
+
+                            // A statement that represents the declarations in a using or fixed statement should not be treated as a separate statement.
+                            switch (declarationSyntax.Parent.Kind())
+                            {
+                                case SyntaxKind.UsingStatement:
+                                    if (declarationSyntax == ((UsingStatementSyntax)declarationSyntax.Parent).Declaration)
+                                    {
+                                        return visited;
+                                    }
+                                    break;
+                                case SyntaxKind.FixedStatement:
+                                    if (declarationSyntax == ((FixedStatementSyntax)declarationSyntax.Parent).Declaration)
+                                    {
+                                        return visited;
+                                    }
+                                    break;
+                            }
+                        }
+                        break;
+                    case BoundKind.MultipleLocalDeclarations:
```

Comment:
That will become much easier to accomplish after switching to the new instrumentation rewriting approach. I'm reluctant to do any instrumentation at the expression/initialization level before we make the change, because doing so would introduce more throwaway code.